src/overlays/actors/ovl_En_Wf/z_en_wf.c

```diff
@@ -137,81 +144,1335 @@ static ColliderCylinderInit D_80B37A5C = {
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wf/func_80B35024.s")
+    if (this->eyeIndex == 0) {
+        if ((Rand_ZeroOne() < 0.2f) && ((globalCtx->gameplayFrames % 4) == 0) && (this->actor.colorFilterTimer == 0)) {
+            this->eyeIndex++;
+        }
+    } else {
+        this->eyeIndex = (this->eyeIndex + 1) & 3;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wf/func_80B35540.s")
+s32 EnWf_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, void* thisx) {
+    EnWf* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wf/func_80B355BC.s")
+    if ((limbIndex == WOLFOS_LIMB_HEAD) || (limbIndex == WOLFOS_LIMB_EYES)) {
+        rot->y -= this->headRot.y;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wf/func_80B3590C.s")
+    return false;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wf/func_80B359A8.s")
+void EnWf_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, void* thisx) {
+    static Vec3f colliderVec = { 1200.0f, 0.0f, 0.0f };
+    static Vec3f bodyPartVec = { 0.0f, 0.0f, 0.0f };
+    EnWf* this = THIS;
+    s32 bodyPartIndex = -1;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wf/func_80B35B94.s")
+    Collider_UpdateSpheres(limbIndex, &this->colliderSphere);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wf/func_80B35C10.s")
+    if (limbIndex == WOLFOS_LIMB_TAIL) {
+        Vec3f colliderPos;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wf/func_80B35D18.s")
+        bodyPartIndex = -1;
+        Matrix_MultVec3f(&colliderVec, &colliderPos);
+        this->colliderCylinder2.dim.pos.x = colliderPos.x;
+        this->colliderCylinder2.dim.pos.y = colliderPos.y;
+        this->colliderCylinder2.dim.pos.z = colliderPos.z;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wf/func_80B35D90.s")
+    if ((this->fireTimer != 0) || ((this->actor.colorFilterTimer != 0) && (this->actor.colorFilterParams & 0x4000))) {
+        switch (limbIndex) {
+            case WOLFOS_LIMB_EYES:
+                bodyPartIndex = 0;
+                break;
+            case WOLFOS_LIMB_FRONT_RIGHT_LOWER_LEG:
+                bodyPartIndex = 1;
+                break;
+            case WOLFOS_LIMB_FRONT_LEFT_LOWER_LEG:
+                bodyPartIndex = 2;
+                break;
+            case WOLFOS_LIMB_THORAX:
+                bodyPartIndex = 3;
+                break;
+            case WOLFOS_LIMB_ABDOMEN:
+                bodyPartIndex = 4;
+                break;
+            case WOLFOS_LIMB_TAIL:
+                bodyPartIndex = 5;
+                break;
+            case WOLFOS_LIMB_BACK_RIGHT_SHIN:
+                bodyPartIndex = 6;
+                break;
+            case 37: // There is no limb with index this large
+                bodyPartIndex = 7;
```

Comment:
`@bug`, I think. doesn't this mean `this->bodyPartsPos[7]` is never initialized, so a wolfos on fire means fire could draw at 0,0,0 ?

Reply:
indeed
![image](https://user-images.githubusercontent.com/22045047/133623670-c50a1e8f-15e0-4389-b0c6-8d4a8b2d4254.png)

Reply:
Documented.